### PR TITLE
Update OpenROAD

### LINKED ISSUE #459: Crash during antenna check portion of Global Routing
### Description

Title	

### Expected Behavior

No crash

I'm only reporting this so I can merge the OpenROAD update to `main` instead of `dev`

### Environment report

```sh
kernel: Darwin
kernel_version: 23.3.0
supported: True
distro: macOS
distro_version: 14.3.1
python_version: 3.11.6
python_path:
  - /nix/store/a4xz3grhgbnxqrvq8m5f4c29n7pbvhm2-python3.11-openlane/bin
  - /nix/store/d1df4rb3p1bzr4qiwysc966y6l0vq2n8-python3-3.11.6-env/lib/python3.11/site-packages
  - /nix/store/7b0rz3bnx7msw2wawkv1hhn5lqf1b0wi-python3-3.11.6/lib/python311.zip
  - /nix/store/7b0rz3bnx7msw2wawkv1hhn5lqf1b0wi-python3-3.11.6/lib/python3.11
  - /nix/store/7b0rz3bnx7msw2wawkv1hhn5lqf1b0wi-python3-3.11.6/lib/python3.11/lib-dynload
  - /nix/store/7b0rz3bnx7msw2wawkv1hhn5lqf1b0wi-python3-3.11.6/lib/python3.11/site-packages
  - /nix/store/a4xz3grhgbnxqrvq8m5f4c29n7pbvhm2-python3.11-openlane/lib/python3.11/site-packages
  - /nix/store/dlq68dqc1i3bsa9nmixfj84jhhdww8s8-python3.11-click-8.1.7/lib/python3.11/site-packages
  - /nix/store/99yxyy0j55h4121bysm73qk71b6bc5id-python3.11-cloup-3.0.2/lib/python3.11/site-packages
  - /nix/store/b7n62zzklq7kwih00k14vhmlvfaqnqk8-python3.11-pyyaml-6.0.1/lib/python3.11/site-packages
  - /nix/store/73bmh8jl0c9bsjha92f7yj6xs3rld58l-python3.11-rich-13.5.2/lib/python3.11/site-packages
  - /nix/store/nb1d926r1dcanqqwg8c7fb2hsgd018mf-python3.11-markdown-it-py-3.0.0/lib/python3.11/site-packages
  - /nix/store/5n74gflis0pknmy1gdr30ys5lnnf10gp-python3.11-mdurl-0.1.2/lib/python3.11/site-packages
  - /nix/store/0ylrh8g1b8yf26094vb7scycrv9h3s4i-python3.11-pygments-2.16.1/lib/python3.11/site-packages
  - /nix/store/wagh5w3zvny6bpknvzrs8gkn6bl80bdk-python3.11-setuptools-68.2.2/lib/python3.11/site-packages
  - /nix/store/zyk45yx79i27d3x984av7mv2p0mjzrh3-python3.11-requests-2.31.0/lib/python3.11/site-packages
  - /nix/store/9v9c8xdsycxhcjzqni3l3yvvnjxrxsf4-python3.11-brotlicffi-1.1.0.0/lib/python3.11/site-packages
  - /nix/store/hj5wvs8agv2ih3s2jznycrycs4yz6c96-python3.11-cffi-1.16.0/lib/python3.11/site-packages
  - /nix/store/s8qds6w7mwzylpxs5gafislgl3gjm6d6-python3.11-pycparser-2.21/lib/python3.11/site-packages
  - /nix/store/7smhm8r5jfn3rh5lpgbmh9z5r5czhv67-python3.11-certifi-2023.07.22/lib/python3.11/site-packages
  - /nix/store/jszrzl2kinjar1738wcbjc1dakdgsi2i-python3.11-charset-normalizer-3.2.0/lib/python3.11/site-packages
  - /nix/store/xq5cvbkv5wns03cz1sflc8wxxs32pgh8-python3.11-idna-3.4/lib/python3.11/site-packages
  - /nix/store/kdasjlza6j1ah8xf3maz55x38mjpf0x1-python3.11-urllib3-2.0.7/lib/python3.11/site-packages
  - /nix/store/1iyvbhygdy6fhzi5xkmvqzll5w9zc77j-python3.11-pcpp-1.30/lib/python3.11/site-packages
  - /nix/store/hksls9h4252wrbla3hbp1prrcxjik3sd-python3.11-volare/lib/python3.11/site-packages
  - /nix/store/xrvvv8gmxy8wv0yrcz71pl0x3dp5nc54-python3.11-httpx-0.25.0/lib/python3.11/site-packages
  - /nix/store/3l3fvl7is0v8hxvjm27g5ww7sygrns8j-python3.11-httpcore-0.18.0/lib/python3.11/site-packages
  - /nix/store/g9xk8r1xh4y1h32gmkv6islzdkx8p58s-python3.11-anyio-4.0.0/lib/python3.11/site-packages
  - /nix/store/w7v0cbsc9psb42aq99igdma14lcf7z0p-python3.11-sniffio-1.3.0/lib/python3.11/site-packages
  - /nix/store/wrkcqllk1cf0g1ff01vwnw4kqd7zbyvx-python3.11-h11-0.14.0/lib/python3.11/site-packages
  - /nix/store/6z1iabfc9hfvak0616p6mfchgjd94sh0-python3.11-rfc3986-2.0.0/lib/python3.11/site-packages
  - /nix/store/vfrmqvhm1ri7c70jlcrq9l4nwz834kp8-python3.11-zstandard-0.21.0/lib/python3.11/site-packages
  - /nix/store/ibrppdxj13agxkiaq0h4nv8nzi10z9ia-python3.11-tkinter-3.11.6/lib/python3.11/site-packages
  - /nix/store/hl2x2jmb4a9bm9d8ms0yyslda5n4185d-python3.11-lxml-4.9.3-3/lib/python3.11/site-packages
  - /nix/store/316zkf7i3b7mha8s8ml8nzzas0j65jjg-python3.11-deprecated-1.2.14/lib/python3.11/site-packages
  - /nix/store/z6aas64929llpbi6qm13b8smr7rq03ld-python3.11-wrapt-1.14.1/lib/python3.11/site-packages
  - /nix/store/sz5x2fxdkm9lan7pii9i2ff7m8b4a26s-python3.11-immutabledict-3.0.0/lib/python3.11/site-packages
  - /nix/store/m731h4hz46dswmchvzd7d2msz6rcw3xz-python3.11-libparse/lib/python3.11/site-packages
  - /nix/store/lvj09754cilhdawl0mhmayl5i81kbfyq-python3.11-psutil-5.9.6/lib/python3.11/site-packages
  - /nix/store/0d9zkk04nbyqlpnxr3icj0hr0qv7nvxs-klayout-pymod/lib/python3.11/site-packages
tkinter: True
container_info:
  engine: UNKNOWN
  version: UNKNOWN
  conmon: False
  rootless: False
nix_info:
  version_string: nix (Nix) 2.18.1
  channels:
    darwin: https://github.com/LnL7/nix-darwin/archive/master.tar.gz
  nix_command: True
  flakes: True
```

### Reproduction material

[repro.tgz](https://github.com/efabless/openlane2/files/15092057/repro.tgz)

### Relevant log output

```sh
Stack trace:
 0# handler(int) in /nix/store/scdc91c8y8qvivsav5dnn9y4phlflnqk-openroad/bin/.openroad-wrapped
 1# _sigtramp in /usr/lib/system/libsystem_platform.dylib
 2# odb::dbWirePathItr::getNextShape(odb::dbWirePathShape&) in /nix/store/scdc91c8y8qvivsav5dnn9y4phlflnqk-openroad/bin/.openroad-wrapped
 3# odb::tmg_conn::loadWire(odb::dbWire*) in /nix/store/scdc91c8y8qvivsav5dnn9y4phlflnqk-openroad/bin/.openroad-wrapped
 4# odb::tmg_conn::analyzeNet(odb::dbNet*) in /nix/store/scdc91c8y8qvivsav5dnn9y4phlflnqk-openroad/bin/.openroad-wrapped
 5# odb::orderWires(utl::Logger*, odb::dbNet*) in /nix/store/scdc91c8y8qvivsav5dnn9y4phlflnqk-openroad/bin/.openroad-wrapped
 6# grt::RepairAntennas::makeNetWire(odb::dbNet*, std::__1::vector<grt::GSegment, std::__1::allocator<grt::GSegment>>&, std::__1::map<int, odb::dbTechVia*, std::__1::less<int>, std::__1::allocator<std::__1::pair<int const, odb::dbTechVia*>>>&) in /nix/store/scdc91c8y8qvivsav5dnn9y4phlflnqk-openroad/bin/.openroad-wrapped
 7# grt::RepairAntennas::makeNetWires(std::__1::map<odb::dbNet*, std::__1::vector<grt::GSegment, std::__1::allocator<grt::GSegment>>, grt::cmpById, std::__1::allocator<std::__1::pair<odb::dbNet* const, std::__1::vector<grt::GSegment, std::__1::allocator<grt::GSegment>>>>>&, int) in /nix/store/scdc91c8y8qvivsav5dnn9y4phlflnqk-openroad/bin/.openroad-wrapped
 8# ant::AntennaChecker::checkAntennas(odb::dbNet*, bool) in /nix/store/scdc91c8y8qvivsav5dnn9y4phlflnqk-openroad/bin/.openroad-wrapped
 9# _wrap_check_antennas(void*, Tcl_Interp*, int, Tcl_Obj* const*) in /nix/store/scdc91c8y8qvivsav5dnn9y4phlflnqk-openroad/bin/.openroad-wrapped
10# Tcl_EvalObjv in /nix/store/l1rhkhykl3vzqipkk1c7wm6qanswxnrk-tcl-8.6.13/lib/libtcl8.6.dylib
11# TclEvalEx in /nix/store/l1rhkhykl3vzqipkk1c7wm6qanswxnrk-tcl-8.6.13/lib/libtcl8.6.dylib
12# Tcl_Eval in /nix/store/l1rhkhykl3vzqipkk1c7wm6qanswxnrk-tcl-8.6.13/lib/libtcl8.6.dylib
13# sta::sourceTclFile(char const*, bool, bool, Tcl_Interp*) in /nix/store/scdc91c8y8qvivsav5dnn9y4phlflnqk-openroad/bin/.openroad-wrapped
14# ord::tclAppInit(Tcl_Interp*) in /nix/store/scdc91c8y8qvivsav5dnn9y4phlflnqk-openroad/bin/.openroad-wrapped
15# Tcl_MainEx in /nix/store/l1rhkhykl3vzqipkk1c7wm6qanswxnrk-tcl-8.6.13/lib/libtcl8.6.dylib
16# main in /nix/store/scdc91c8y8qvivsav5dnn9y4phlflnqk-openroad/bin/.openroad-wrapped
```

### PR DESCRIPTION
**Pull request by donn**
## Tool Updates
* Updated OpenROAD to `d423155`, OpenSTA to `a7f3421`
  * Addresses an [antenna repair bug](https://github.com/efabless/openlane2/issues/459)

## Testing
* Updated a number of unit tests to reflect new OpenROAD error codes.
* Fixed failing design integration tests.

---

Depends on https://github.com/efabless/openlane2-step-unit-tests/pull/29
Depends on https://github.com/efabless/openlane2-ci-designs/pull/26
Resolves #459

### CODE REVIEW

**Comment by kareefardi** [comment]
@donn Is there an open issue for the bug?

**Comment by donn** [comment]
@kareefardi Nope. I would forgo opening one since tracking down a test case is going to be annoying.

That said it looks like we have bigger problems.

**Comment by kareefardi** [comment]
> @kareefardi Nope. I would forgo opening one since tracking down a test case is going to be annoying.
> 
> That said it looks like we have bigger problems.

Would have been nice to find the test case and see if OR added it to their regression and add it to ours if they didn't.

File: openlane/steps/openroad.py [hunk 1]
```diff
@@ -225,7 +225,7 @@ def on_alert(self, alert: OpenROADAlert) -> OpenROADAlert:
         if alert.code in [
             "ORD-0039",  # .openroad ignored with -python
             "ODB-0220",  # lef parsing/NOWIREEXTENSIONATPIN statement is obsolete in version 5.6 or later.
-            "STA-0122",  # table template \w+ not found
+            "STA-1256",  # table template \w+ not found
```
Comment (kareefardi) [inline openlane/steps/openroad.py:228]: 🙄

**Review by kareefardi — APPROVED** [review]

File: openlane/steps/openroad.py (hunk 1 above)
Comment (donn) [reply openlane/steps/openroad.py:228]: yep…

**Pull request merged**